genesyscloud/resource_genesyscloud_knowledge_category.go

```diff
@@ -0,0 +1,281 @@
+			},
+		},
+	}
+)
+
+func getAllKnowledgeCategories(_ context.Context, clientConfig *platformclientv2.Configuration) (ResourceIDMetaMap, diag.Diagnostics) {
+	knowledgeBaseList := make([]platformclientv2.Knowledgebase, 0)
+	resources := make(ResourceIDMetaMap)
+	knowledgeAPI := platformclientv2.NewKnowledgeApiWithConfig(clientConfig)
+
+	for pageNum := 1; ; pageNum++ {
+		const pageSize = 100
+		knowledgeBases, _, getErr := knowledgeAPI.GetKnowledgeKnowledgebases("", "", "", fmt.Sprintf("%v", pageSize), "", "", false, "", "")
+		if getErr != nil {
+			return nil, diag.Errorf("Failed to get page of knowledge bases: %v", getErr)
+		}
+
+		if knowledgeBases.Entities == nil || len(*knowledgeBases.Entities) == 0 {
+			break
+		}
+		for _, knowledgeBase := range *knowledgeBases.Entities {
+			knowledgeBaseList = append(knowledgeBaseList, knowledgeBase)
+		}
+	}
+	for _, knowledgeBase := range knowledgeBaseList {
+		for pageNum := 1; ; pageNum++ {
+			const pageSize = 100
+			knowledgeCategories, _, getErr := knowledgeAPI.GetKnowledgeKnowledgebaseLanguageCategories(*knowledgeBase.Id, *knowledgeBase.CoreLanguage, "", "", "", fmt.Sprintf("%v", pageSize), "")
+			if getErr != nil {
+				return nil, diag.Errorf("Failed to get page of knowledge categories: %v", getErr)
+			}
+
+			if knowledgeCategories.Entities == nil || len(*knowledgeCategories.Entities) == 0 {
+				break
+			}
+
+			for _, knowledgeCategory := range *knowledgeCategories.Entities {
+				id := fmt.Sprintf("%s %s %s", *knowledgeCategory.Id, *knowledgeCategory.KnowledgeBase.Id, *knowledgeCategory.LanguageCode)
+				resources[id] = &ResourceMeta{Name: *knowledgeCategory.Name}
+			}
+		}
+	}
+
+	return resources, nil
+}
+
+func knowledgeCategoryExporter() *ResourceExporter {
+	return &ResourceExporter{
+		GetResourcesFunc: getAllWithPooledClient(getAllKnowledgeCategories),
+		RefAttrs:         map[string]*RefAttrSettings{}, // No references
+	}
+}
+
+func resourceKnowledgeCategory() *schema.Resource {
+	return &schema.Resource{
+		Description: "Genesys Cloud Knowledge Category",
+
+		CreateContext: createWithPooledClient(createKnowledgeCategory),
+		ReadContext:   readWithPooledClient(readKnowledgeCategory),
+		UpdateContext: updateWithPooledClient(updateKnowledgeCategory),
+		DeleteContext: deleteWithPooledClient(deleteKnowledgeCategory),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"knowledge_base_id": {
+				Description: "Knowledge base id of the category",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"language_code": {
+				Description:  "language code of the category",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"en-US", "en-UK", "en-AU", "de-DE", "es-US", "es-ES", "fr-FR", "pt-BR", "nl-NL", "it-IT", "fr-CA"}, false),
+			},
+			"knowledge_category": {
```

Comment:
I don't think the name and description of the knowledgeCategory element would have any effect on the resource. It's probably best to name this something like `parent_knowledge_category_id` and just have it as a string value containing an ID.
Will need to include it in the tests too to make sure it works

Reply:
knowledgeCategory is actually defining the name and description of the category being created, not those of the parent category.

Reply:
Oh sorry I misread that